lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -428,65 +428,83 @@ def _call(self, method: Callable, json_data: Dict[AnyStr, Any]):
         Returns:
             Postprocessed JSON body to return to client
         """
+        klasname = self.__class__.__name__
         try:
             es_request = self.assemble(json_data)
             path = es_request.get("path")
             url = urljoin(self.es_url, path)
         except Exception as e:
-            self.logger.exception("Blew it in setup: {}", type(e).__name__)
+            self.logger.exception("{} blew it in setup: {}", klasname, type(e).__name__)
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
         try:
             # query Elasticsearch
             es_response = method(url, **es_request["kwargs"])
             es_response.raise_for_status()
         except requests.exceptions.HTTPError as e:
-            self.logger.exception("HTTP error {} from Elasticsearch request", e)
+            self.logger.exception(
+                "{} HTTP error {} from Elasticsearch request", klasname, e
+            )
+            # FIXME: I can't seem to figure out how to give the mock exception
+            # the proper attributes; so rather than have this blow up, or
+            # remove the information for production deployment, simply avoid
+            # trying to access them in unit tests.
```

Comment:
I think you can set these attributes via `request` and `response` keyword arguments to the `requests.exceptions.HTTPError` c'tor.  Hopefully, that will allow you to removed this stuff from here.

Reply:
Yes, you can; however I had trouble figuring out a way to construct a useful `Response` object in order to do that; the constructor, it turns out, doesn't allow specifying attributes. You have to create it and then set them later, which in this context is awkward. It did occur to me to try just creating a class `HTTPError` with a Response object and using that here instead of specifying the class name, although it's not clear from documentation exactly what the requests mock's assumptions are regarding this parameter; in the end I decided just to cut the mess short and get on with it. As frequently happens, you make me question that again ...

This, by the way, came out of fixing another instance of a broken f-string... we never cared before because, although I'd included the parameters, they were never interpreted! I really hate that I made a trivial single character fix, and broke all my unit tests in a way that proved exceedingly difficult to work around and harder to fix.

Reply:
> I really hate that I made a trivial single character fix, and broke all my unit tests

:-D  Oh, I'm so sorry!  :-}

Anyway, since this is all based on the Python `requests` package, can't you use the `responses` package to mock the response from the call to ES?  That seems like it should allow you do drive the behavior of the code in question from "outside" without having to make special accommodation for it being run in a unit test.

Reply:
In theory, yes; except the issue here is the `exc=` kwarg we're using to mock a failure exception, and there's no special help I can find in constructing this. It's normally shown as just the class name, and we're not given a properly constructed instance of that type so it causes a mess.

The new commit uses a function to construct a properly attributed exception, but it's a bit ugly. It's more ugly because there's no real way to do this in a common fixture, so it's duplicated. I really want to find some time to refactor the query_api tests as a class hierarchy as I did for the implementation ... it's just not made it to the top of my list. (Although this PR pushed my patience a bit, I really didn't want to take the extra time that would have required before getting up the dataset changes and other fixes here.)